proofgen/imports.go

```diff
@@ -55,4 +55,29 @@ func translateImports(w io.Writer, pkg *packages.Package, usingFfi bool, ffi str
 	for _, imp := range tr.importsList {
 		fmt.Fprintf(w, "Require Export New.generatedproof.%s.\n", imp)
 	}
+	return tr
+}
+
+func importPackageIdentifier(pkgPath string) string {
+	i := strings.LastIndex(pkgPath, ".")
+	if i < 0 {
+		return pkgPath
+	}
+	return pkgPath[i+1:]
+}
+
+func (tr *importsTranslator) translateImportList(w io.Writer, pkg *packages.Package) {
```

Comment:
At some point, it might be better to emit a list of imported packages in the trusted Goose translation, which would eliminate the need for this. The Go code's translation currently inlines calls to `importA.initialize'`, `importB.initialize'` and so forth into the current package's `initialize` function, which prevents any abstraction in Coq about the import phase of initializing a package.

Reply:
The list of imported packages is now in trusted code. Initialization uses `globals.package_init` which now takes most of its parameters through the PkgInfo class, but the translator still produces a function that calls all the dependencies - as with `is_defined`, abstracting this out is hard since we would need the `PkgInfo` of every dependent package and this isn't easy to abstract over.